### ISSUE TITLE
maxAge doesn't work as expected

### ISSUE THREAD

**Issue by mathiasose** [auth0/node-jsonwebtoken#609]
I have made three tests at https://runkit.com/mathiasose/5d0901da2286fc001a4bb01d

I would expect all three to throw an error with `maxAge exceeded`, but only the third one does? Can anyone explain? All the values are expressed in seconds right?

**Comment by gallagherrchris** [comment]
`maxAge` is simply implemented as an additional way to limit the length of time a token is valid.  This allows you to not accept a token even if it is not expired yet.  In your first example, the time between the `iat` and when you verify is less than the `maxAge` therefore it successfully validates. 

 You can check out the code here
https://github.com/auth0/node-jsonwebtoken/blob/eefb9d9c6eec54718fa6e41306bda84788df7bec/verify.js#L199-L211

**Comment by panva** [comment]
> const token1 = jwt.sign({test: 1}, key, {expiresIn: '10h'});
> const payload1 = jwt.verify(token1, key, {maxAge: maxAge});
> console.log(payload1);

This token when signed has an `iat` of now() and `exp` of 10hours from now. Validation `maxAge` is set to one hour so if you waited for one hour, regardless of the `exp` the option would kick in an throw.

> const token2 = jwt.sign({test: 2, iat: 1, exp: 10 * 60 * 60}, key, {});
> const payload2 = jwt.verify(token2, key, {maxAge: maxAge, clockTimestamp: 1});
> console.log(payload2);

This token when signed has an `iat` of 1, `exp` of 36000. Validation `clockTimestamp` is set to `1` so as it's validating it thinks the current timestamp is `1`, hence the token is not expired yet and never would be given these validate options.

> const token3 = jwt.sign({test: 3, iat: 1000000000, exp: 9999999999}, key, {});
> const payload3 = jwt.verify(token3, key, {maxAge: maxAge});
> console.log(payload3);

This token's iat is equal to date `2001-09-09 03:46:40 +0200`, exp is far in the future but `maxAge` only set to 1h, hence, since 2001-09-09 is way further than 1 hour ago. It throws.

Bottom line, `maxAge` option throws WHEN iat + maxAge >= now. `clockTimestamp` fixes the now in place to a specific value.

Hope it makes sense.
